src/Css.elm

```diff
@@ -854,21 +855,25 @@ tools which express these as e.g. `#abcdef0`, etc.
 -}
 hex : String -> Color
 hex str =
-  -- TODO emit warnings for invalid hex chars
   let
     value =
       if String.slice 0 1 str == "#" then
         str
       else
         "#" ++ str
+    warnings =
+      if contains (regex "^#([a-fA-F0-9]{6}|[a-fA-F0-9]{3})$") value then
+        [ "Hexadecimal color values can only contain three or six characters of '0-9' and/or 'A-F'. " ++ value ++ " is not valid." ]
```

Comment:
current CSS4 draft [supports lengths of 4 and 8](https://drafts.csswg.org/css-color/#hex-notation) (for alpha); I'm inclined to think allowing those is okay. 😄